drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -469,13 +469,13 @@ static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
 
     switch (opt) {
         case NETOPT_ADDRESS:
-            assert(len <= sizeof(uint16_t));
-            at86rf2xx_set_addr_short(dev, *((const uint16_t *)val));
+            assert(len == sizeof(network_uint16_t));
```

Comment:
The expected type for `NETOPT_ADDRESS` is still a byte array, so the assert should be

```suggestion
            assert(len >= sizeof(network_uint16_t));
```